### PR TITLE
Add Support for HW that indicates armed state.

### PR DESCRIPTION
**Pull request by davids5**
Boards may choose to support this feature by defining the macro BOARD_INDICATE_ARMED_STATE in their board_config.h file.

fmuv5: Repurpose TIM5_SPARE_4 as nARMED 
On V5 nARMED is a Digital OUTPUT. GPIO will be set as input while not armed HW will have Pull UP. While armed it will be configured as a GPIO OUT set LOW.

image

### CODE REVIEW

File: boards/px4/fmu-v5/src/board_config.h [hunk 1]
```diff
@@ -347,9 +347,14 @@
...
 #define DIRECT_PWM_CAPTURE_CHANNELS  3
 
-/* TIM5_CH4 SPARE PIN */
-#define GPIO_TIM5_CH4IN    /* PI0   T5C4  TIM5_SPARE_4 */  GPIO_TIM5_CH4IN_2
-#define GPIO_TIM5_CH4OUT   /* PI0   T5C4  TIM5_SPARE_4 */   GPIO_TIM5_CH4OUT_2
+/* PI0 is nARMED
+ *  The GPIO will be set as input while not armed HW will have external HW Pull UP.
+ *  While armed it shall be configured at a GPIO OUT set LOW
+ */
+#define GPIO_nARMED_INIT     /* PI0 */  (GPIO_INPUT |GPIO_PULLUP|GPIO_PORTI|GPIO_PIN1)
```
Comment (dagar) [inline boards/px4/fmu-v5/src/board_config.h]:
```suggestion
#define GPIO_nARMED_INIT     /* PI0 */  (GPIO_INPUT|GPIO_PULLUP|GPIO_PORTI|GPIO_PIN1)
```

**Comment by davids5** [comment]
@dagar fixed up and rebased on master

**Comment by dagar** [comment]
CI fail.

image

**Comment by davids5** [comment]
@dagar so much for editing online...did not see the ws.

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -2687,6 +2687,10 @@ control_status_leds(vehicle_status_s *status_local, const actuator_armed_s *actu
 
 	last_overload = overload;
 
+	/* board supports HW armed indicator */
+
+	BOARD_INDICATE_ARMED_STATE(actuator_armed->armed);
```
Comment (dagar) [inline src/modules/commander/Commander.cpp]: Only on change?

**Pull request merged**